lib/Sabberworm/CSS/Parser.php

```diff
@@ -26,22 +26,25 @@
  */
 class Parser {
 
-	private $sText;
+	private $aText;
 	private $iCurrentPosition;
 	private $oParserSettings;
 	private $sCharset;
 	private $iLength;
-	private $peekCache = null;
 	private $blockRules;
 	private $aSizeUnits;
 
 	public function __construct($sText, Settings $oParserSettings = null) {
-		$this->sText = $sText;
 		$this->iCurrentPosition = 0;
 		if ($oParserSettings === null) {
 			$oParserSettings = Settings::create();
 		}
 		$this->oParserSettings = $oParserSettings;
+		if ($this->oParserSettings->bMultibyteSupport) {
+			$this->aText = preg_split('//u', $sText, null, PREG_SPLIT_NO_EMPTY);
```

Comment:
I’m not sure `preg_split` is multibyte-safe. Sure it supports utf-8 (if compiled in) but I don’t think it copes well with other encodings like utf-16, shift-jis. Also there is no way to pass the encoding used. I’ve never added unit tests for other encodings, though, because the identifier parsing code wasn’t actually able to match any encoding where the identifiers weren’t ASCII. Maybe I’ll just make that my next big rework project…